src/main/java/org/craftercms/studio/impl/v2/repository/blob/BlobAwareContentRepository.java

```diff
@@ -0,0 +1,540 @@
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Implementation of {@link ContentRepository}, {@link org.craftercms.studio.api.v2.repository.ContentRepository} and
+ * {@link DeploymentHistoryProvider} that delegates calls to a {@link BlobStore} when appropriate
+ *
+ * @author joseross
+ * @since 3.1.6
+ */
+public class BlobAwareContentRepository implements ContentRepository, DeploymentHistoryProvider,
+        org.craftercms.studio.api.v2.repository.ContentRepository {
+
+    private static final Logger logger = LoggerFactory.getLogger(BlobAwareContentRepository.class);
+
+    /**
+     * The extension for the blob files
+     */
+    protected String fileExtension;
+
+    protected GitContentRepository localRepositoryV1;
+
+    protected org.craftercms.studio.impl.v2.repository.GitContentRepository localRepositoryV2;
+
+    protected BlobStoreFactory factory;
+
+    protected ObjectMapper objectMapper = new XmlMapper().enable(SerializationFeature.INDENT_OUTPUT);
+
+    public void setFileExtension(String fileExtension) {
+        this.fileExtension = fileExtension;
+    }
+
+    public void setLocalRepositoryV1(GitContentRepository localRepositoryV1) {
+        this.localRepositoryV1 = localRepositoryV1;
+    }
+
+    public void setLocalRepositoryV2(org.craftercms.studio.impl.v2.repository.GitContentRepository localRepositoryV2) {
+        this.localRepositoryV2 = localRepositoryV2;
+    }
+
+    public void setFactory(BlobStoreFactory factory) {
+        this.factory = factory;
+    }
+
+    protected boolean isFolder(String path) {
+        return StringUtils.isEmpty(FilenameUtils.getExtension(path));
+    }
+
+    protected String getOriginalPath(String path) {
+        return StringUtils.removeEnd(path, "." + fileExtension);
+    }
+
+    protected String getPointerPath(String path) {
+        return isFolder(path)? path : StringUtils.appendIfMissing(path, "." + fileExtension);
+    }
+
+    protected BlobStore getRemoteStore(String site, String... paths)
+            throws ServiceLayerException, ConfigurationException {
+        if (ArrayUtils.isEmpty(paths)) {
+            throw new IllegalArgumentException("At least one path needs to be provided");
+        }
+        return factory.getByPaths(site, paths);
+    }
+
+    // Start API 1
+
+    @Override
+    public boolean contentExists(String site, String path) {
+        logger.debug("Checking if {0} exists in site {1}", path, site);
+        try {
+            if (!isFolder(path)) {
+                BlobStore store = getRemoteStore(site, path);
+                if (store != null) {
+                    return store.contentExists(site, path);
+                }
+            }
+            return localRepositoryV1.contentExists(site, path);
+        } catch (Exception e) {
+            return false;
```

Comment:
Not logging exceptions?

Reply:
This is still not resolved. Can you explain why are you not logging the exception?